### PR TITLE
Unifying multiBar and multiBarHorizontal

### PR DESCRIPTION
**Pull request by jozefvaclavik**
I'm afraid I did lot of refactoring to make both models equal.

Whats new:
* both models now include same set of features
* uses same logic to draw content
* yErr/xErr now display within chart
* show values in combination with yErr/xErr now display value outside of yErr/xErr

What would be nice to fix
* values can be still displayed outside of chart. I guess it's impossible to calculate length of text
* multiBar xAxis is drawn with padding and therefore values are displayed across the text - first screenshot Group A, Group B, etc... I couldn't figure out how to make it stick to the bottom. My guess would be that domain instead of range needs to be changed..

Here are some screens of a chart once you change model from multiBar to multiBarHorizontal
screen shot 2015-01-18 at 14 20 10
screen shot 2015-01-18 at 14 19 51
screen shot 2015-01-18 at 14 19 58
screen shot 2015-01-18 at 14 20 19

Yeah and sorry for all the commits. Last minute bugs :disappointed:

### CODE REVIEW

File: src/models/multiBar.js [hunk 1]
```diff
@@ -104,29 +99,44 @@ nv.models.multiBar = function() {
...
-            // If scale's domain don't have a range, slightly adjust to make one... so a chart can show a single data point
-            if (x.domain()[0] === x.domain()[1])
-                x.domain()[0] ?
-                    x.domain([x.domain()[0] - x.domain()[0] * 0.01, x.domain()[1] + x.domain()[1] * 0.01])
-                    : x.domain([-1,1]);
-
-            if (y.domain()[0] === y.domain()[1])
-                y.domain()[0] ?
-                    y.domain([y.domain()[0] + y.domain()[0] * 0.01, y.domain()[1] - y.domain()[1] * 0.01])
-                    : y.domain([-1,1]);
```
Comment (jozefvaclavik) [inline src/models/multiBar.js]: I didn't work yet on this feature. I'll push it with next commit.

**Comment by liquidpele** [comment]
Sorry for the delay, will try to look at this when I can... doing home remodeling :/

File: src/models/multiBar.js [hunk 2]
```diff
@@ -247,49 +321,32 @@ nv.models.multiBar = function() {
...
 
+            }
             //store old scales for use in transitions on update
             x0 = x.copy();
             y0 = y.copy();
 
-            // keep track of the last data value length for transition calculations
-            if (data[0] && data[0].values) {
-                last_datalength = data[0].values.length;
-            }
```
Comment (liquidpele) [inline src/models/multiBar.js]: Can't remove this, it fixed another bug.

File: src/models/multiBar.js [hunk 3]
```diff
@@ -35,11 +40,8 @@ nv.models.multiBar = function() {
     // Private Variables
     //------------------------------------------------------------
 
-    var x0, y0 //used to store previous scales
-        , renderWatch = nv.utils.renderWatch(dispatch, duration)
-        ;
-
-    var last_datalength = 0;
```
Comment (liquidpele) [inline src/models/multiBar.js]: Can't remove this, it fixed another bug.

File: src/models/multiBar.js [hunk 4]
```diff
@@ -9,25 +9,30 @@ nv.models.multiBar = function() {
...
-        , id = Math.floor(Math.random() * 10000) //Create semi-unique ID in case user doesn't select one
         , getX = function(d) { return d.x }
         , getY = function(d) { return d.y }
+        , getXerr = function(d) { return d.xErr }
         , forceY = [0] // 0 is forced by default.. this makes sense for the majority of bar graphs... user can always do chart.forceY([]) to remove
-        , clipEdge = true
-        , stacked = false
-        , stackOffset = 'zero' // options include 'silhouette', 'wiggle', 'expand', 'zero', or a custom function
         , color = nv.utils.defaultColor()
-        , hideable = false
```
Comment (liquidpele) [inline src/models/multiBar.js]: Why remove this option?

File: src/models/multiBar.js [hunk 5]
```diff
@@ -152,43 +175,33 @@ nv.models.multiBar = function() {
...
-                .attr('height', 0)
+            groups.exit().watchTransition(renderWatch, 'multibar: exit groups')
+                .style('stroke-opacity', 1e-6)
+                .style('fill-opacity', 1e-6)
                 .remove();
-            if (exitTransition.delay)
-                exitTransition.delay(function(d,i) {
-                    var delay = i * (duration / (last_datalength + 1)) - i;
-                    return delay;
-                });
```
Comment (liquidpele) [inline src/models/multiBar.js]: Why did you remove the transition code?

**Comment by liquidpele** [comment]
There seems to be some major changes here... so much changed it's hard to tell but it looks like some functionality is getting lost here.  Can you build the files in your repo so I can use them in jsfiddle or something to test it out?

**Comment by jozefvaclavik** [comment]
No problem. I can add removed stuff into both charts. I thought that they were fixing something, but these changes were in one of both charts which didn't make sence why they would be only in one.

Give me 2-3 days (or maybe a week :sweat_smile: ) and I'll work it back and prepare some jsfiddle for testing.

**Comment by liquidpele** [comment]
Cool, thanks for contributing :)

**Comment by jozefvaclavik** [comment]
Hey man, I'm sorry to say but I won't have time to finish these changes. Honestly we moved to different charting library. The inconsistency between charts in nvd3 was too strong. If you would like continue with this PR you can take over it and finish it.

**Comment by liquidpele** [comment]
No worries.  What did you move to if you dont mind me asking?

**Comment by jozefvaclavik** [comment]
I made you collaborator in my repo.

Anyway, we end up with Highcharts. They have it's own upsides and downsides as well, but it was better fit for our project.

**Comment by liquidpele** [comment]
Ah, yep highcharts is nice!  (if you're open source or have the $ of course)

**Comment by liquidpele** [comment]
This looks to be too out of date to merge now.  Closing, and if you'd still like to take this on then please submit a new pull request to the novus branch.  Thanks!
